ivadomed/loader/utils.py

```diff
@@ -520,3 +522,116 @@ def reorient_image(arr, slice_axis, nib_ref, nib_ref_canonical):
+    df = layout.to_df(metadata=True)
+
+    # Add filename and parent_path columns
+    df['filename'] = df['path'].apply(os.path.basename)
+    df['parent_path'] = df['path'].apply(os.path.dirname)
+
+    # Filter rows with chosen extensions and data_types from loader parameters
+    df = df[df['filename'].str.endswith(tuple(extensions))]
+
+    # Update dataframe with subject files with chosen contrast and derivative files with chosen target_suffix from loader parameters
+    df = (df[(~df['path'].str.contains('derivatives') & df['suffix'].str.contains('|'.join(contrast_lst))) |
+          df['filename'].str.contains('|'.join(target_suffix))])
+
+    # Add metadata from participants.tsv file, if present
+    # Uses pybids function
+    if layout.get_collections(level='dataset'):
+        df_participants = layout.get_collections(level='dataset', merge=True).to_df()
+        df_participants.drop(['suffix'], axis=1, inplace=True)
+        df = pd.merge(df, df_participants, on='subject', suffixes=("_x", None), how='left')
+
+    # Add metadata from samples.tsv file, if present
+    # TODO: use pybids function after BEP microscopy is merged in BIDS
+    fname_samples = os.path.join(bids_path, "samples.tsv")
+    if os.path.exists(fname_samples):
+        df_samples = pd.read_csv(fname_samples, sep='\t')
+        df['participant_id'] = "sub-" + df['subject']
+        df['sample_id'] = "sample-" + df['sample']
+        df = pd.merge(df, df_samples, on=['participant_id', 'sample_id'], suffixes=("_x", None), how='left')
+        df.drop(['participant_id', 'sample_id'], axis=1, inplace=True)
+
+    # Add metadata from all _sessions.tsv files, if present
+    # Uses pybids function
+    if layout.get_collections(level='subject'):
+        df_sessions = layout.get_collections(level='subject', merge=True).to_df()
+        df_sessions.drop(['suffix'], axis=1, inplace=True)
+        df = pd.merge(df, df_sessions, on=['subject', 'session'], suffixes=("_x", None), how='left')
+
+    # Add metadata from all _scans.tsv files, if present
+    # TODO: use pybids function after BEP microscopy is merged in BIDS
+    # TODO: verify merge behavior with EEG and DWI scans files, tested with anat and microscopy only
+    df_scans = pd.DataFrame()
+    for root, dirs, files in os.walk(bids_path):
+        for file in files:
+            if file.endswith("scans.tsv"):
+                df_temp = pd.read_csv(os.path.join(root, file), sep='\t')
+                df_scans = pd.concat([df_scans, df_temp], ignore_index=True)
+    if not df_scans.empty:
+        df_scans['filename'] = df_scans['filename'].apply(os.path.basename)
+        df = pd.merge(df, df_scans, on=['filename'], suffixes=("_x", None), how='left')
+
+    # TODO: check if other files are needed for EEG
+
+    # If indexing of derivatives is true
+    # Get list of subject files, list of subject prefix filenames without extensions, and list of available derivatives
+    if derivatives:
```

Comment:
as discussed in meeting, you could inform with logger.debug if one derivative for a file is missing

Reply:
Done in 7c3e8a2: warnings for missing one derivative for a subject, and missing all derivatives for a subject.